Sources/SKCore/BuildSystem.swift

```diff
@@ -43,6 +43,27 @@ public struct SourceFileInfo: Sendable {
   }
 }
 
+/// A target / run destination combination. For example, a configured target can represent building the target
+/// `MyLibrary` for iOS.
+public struct ConfiguredTarget: Hashable, Sendable {
+  /// An opaque string that represents the target.
+  ///
+  /// The target's ID should be generated by the build system that handles the target and only interpreted by that
+  /// build system.
+  public let targetID: String
+
+  /// An opaque string that represents the run destination.
+  ///
+  /// The run destination's ID should be generated by the build system that handles the target and only interpreted by
+  /// that build system.
+  public let runDestinationID: String
```

Comment:
Do we need a "configuration" as well, eg. debug/release/etc? Run destination is fairly general, so you could definitely include platform/sdk/architecture/etc in that, but I'm not sure configuration makes much sense there.

Reply:
Discussed offline that we don’t need it for now because the configuration is picked globally when launching sourcekit-lsp. If we do allow changing the configuration while sourcekit-lsp is running, we’ll need to re-consider whether to add configuration here.